spec/controllers/steps/evidence/upload_controller_spec.rb

```diff
@@ -3,40 +3,36 @@
 RSpec.describe Steps::Evidence::UploadController, type: :controller do
   it_behaves_like 'a generic step controller', Steps::Evidence::UploadForm, Decisions::EvidenceDecisionTree
   it_behaves_like 'a step that can be drafted', Steps::Evidence::UploadForm
+
   describe 'additional CRUD actions' do
-    include_context 'with an existing document'
+    let(:crime_application) { CrimeApplication.create }
+    let(:document) { crime_application.documents.create }
 
     context 'deleting a document' do
       it 'has the expected step name' do
-        document.save
-
         allow_any_instance_of(Datastore::Documents::Delete).to receive(:call).and_return(true)
 
         expect(
           subject
         ).to receive(:update_and_advance).with(
           Steps::Evidence::UploadForm, as: :delete_document,
-          flash: { success: 'Document has been sucessfully deleted' },
-          record: document.document_bundle
+          flash: { success: 'Document has been successfully deleted' }
         )
 
-        put :update, params: { id: bundle.crime_application, document_id: document.id }
+        put :update, params: { id: crime_application, document_id: document }
       end
 
       it 'is has the correct flash message when delete is unsuccessful' do
-        document.save
-
         allow_any_instance_of(Datastore::Documents::Delete).to receive(:call).and_return(false)
 
         expect(
           subject
         ).to receive(:update_and_advance).with(
           Steps::Evidence::UploadForm, as: :delete_document,
-          flash: { alert: 'Document was not sucessfully deleted' },
-          record: document.document_bundle
+          flash: { alert: 'Document was not successfully deleted' }
```

Comment:
This might change right? Do you want to reference the actual i18n value

Reply:
Didn't make major changes to what was done already.
In this case as it is a spec I'm not too worry of this having hardcoded copy. If copy changes the spec will fail and is easy to fix.

In any case the whole deletion will need to be refactored/improved as currently does hard delete on S3 and also tries to do so for not stored files (without s3 key) which fails again, and again 🥲